geps/gep-3614/index.md

```diff
@@ -0,0 +1,70 @@
+# GEP-3614: Firewall
+
+* Issue: [#3614](https://github.com/kubernetes-sigs/gateway-api/issues/3614)
+* Status: Provisional
+
+## TLDR
+
+The ability to attach firewall rules for ingress L3, L4 and L7 Gateway traffic.
+
+## Motivation
+
+`Gateways` are commonly exposed to the internet, which puts them as risk of
+attack. Internal networks can become compromised as well. We should provide
+tooling, documentation and best-practices for users to restrict and control
+access to their `Gateways`.
+
+### User Stories
+
+* As an application developer, I want to allow specific IPs to access my
+  application.
+* As an application developer, I want all ingress traffic for my Gateways to be
+  restricted to my CDN.
+* As an application developer, I want to block all requests from specific user
+  agents OR only allow specific user agents.
+* As a gateway operator I want to be able to identify and block malformed HTTP
+  requests before they reach backend applications.
+* As a gateway operator I want to filter/sanitize requests to block XSS attacks
```

Comment:
What about SQLi/RCE/SSRF, etc? I guess if we mention one attack and we leave others out, we might always have an incomplete list.  And we probably need to add more attacks as they appear...?